src/require_context.js

```diff
@@ -0,0 +1,56 @@
+import vm from 'vm';
+import fs from 'fs';
+import path from 'path';
+
+function requireModules(keys, root, directory, regExp, recursive) {
+  const files = fs.readdirSync(path.join(root, directory));
+
+  files.forEach((filename) => {
+    if (regExp.test(filename)) {
+      // webpack adds a './' to the begining of the key
+      // TODO: Check this in windows
+      const entryKey = `./${path.join(directory, filename)}`;
+      // eslint-disable-next-line no-param-reassign, global-require, import/no-dynamic-require
+      keys[entryKey] = require(path.join(root, directory, filename));
+      return;
+    }
+
+    if (!recursive) {
+      return;
+    }
+
+    if (fs.statSync(path.join(root, directory, filename)).isDirectory()) {
+      requireModules(keys, root, path.join(directory, filename), regExp, recursive);
+    }
+  });
+}
+
+export default function runWithRequireContext(content, options) {
+  const { filename, dirname } = options;
+
+  require.context = (directory, useSubdirectories = false, regExp = /^\.\//) => {
+    const fullPath = path.resolve(dirname, directory);
+    const keys = {};
+
+    requireModules(keys, fullPath, '.', regExp, useSubdirectories);
+
+    const req = f => (keys[f]);
+
+    req.keys = () => (Object.keys(keys));
+
+    console.log(keys);
+
+    return req;
+  };
+
+  // the script will use the module instance of this module.
+  const context = {
+    ...global,
+    require,
```

Comment:
It seems its difficult to replace require. We can't run our own code and then call the real require, because it will create a new `require` instance for the newly required module replacing our code. So we have to completely replace `require`. There are lots of stuff going on we will need to handle ourselves. So I think its best if we work on it later after getting this in. I made other minor changes needed.